src/libs/EmojiTrie.js

```diff
@@ -18,26 +18,40 @@ function createTrie(lang = CONST.LOCALES.DEFAULT) {
             return;
         }
 
-        const name = isDefaultLocale ? item.name : _.get(langEmojis, [item.code, 'name']);
-        const names = isDefaultLocale ? [name] : [...new Set([name, item.name])];
-        _.forEach(names, (nm) => {
-            const node = trie.search(nm);
-            if (!node) {
-                trie.add(nm, {code: item.code, types: item.types, name: nm, suggestions: []});
-            } else {
-                trie.update(nm, {code: item.code, types: item.types, name: nm, suggestions: node.metaData.suggestions});
-            }
-        });
+        const defaultName = item.name;
+        const preferredLanguageName = _.get(langEmojis, [item.code, 'name'], defaultName);
 
+        // Add the actual name of the emoji for the current language
+        const node = trie.search(preferredLanguageName);
+        if (!node) {
+            trie.add(preferredLanguageName, {code: item.code, types: item.types, name: preferredLanguageName, suggestions: []});
+        } else {
+            trie.update(preferredLanguageName, {code: item.code, types: item.types, name: preferredLanguageName, suggestions: node.metaData.suggestions});
+        }
+
+        // Add keywords of both the current language and the default language, in case current language isn't the default.
         const keywords = _.get(langEmojis, [item.code, 'keywords'], []).concat(isDefaultLocale ? [] : _.get(localeEmojis, [CONST.LOCALES.DEFAULT, item.code, 'keywords'], []));
         for (let j = 0; j < keywords.length; j++) {
             const keywordNode = trie.search(keywords[j]);
             if (!keywordNode) {
-                trie.add(keywords[j], {suggestions: [{code: item.code, types: item.types, name}]});
+                trie.add(keywords[j], {suggestions: [{code: item.code, types: item.types, name: preferredLanguageName}]});
             } else {
                 trie.update(keywords[j], {
                     ...keywordNode.metaData,
-                    suggestions: [...keywordNode.metaData.suggestions, {code: item.code, types: item.types, name}],
+                    suggestions: [...keywordNode.metaData.suggestions, {code: item.code, types: item.types, name: preferredLanguageName}],
+                });
+            }
+        }
+
+        // If current language isn't the default, prepend the English name of the emoji in the suggestions as well.
```

Comment:
```suggestion
        // Prepend the English name of the emoji in the suggestions in order to...
```
Could you add a brief explanation of why we're doing this?

Reply:
Sure, one moment!